src/core/shutdown.c

```diff
@@ -203,20 +202,25 @@ int main(int argc, char *argv[]) {
         }
 
         (void) cg_get_root_path(&cgroup);
+        in_container = detect_container() > 0;
 
         use_watchdog = !!getenv("WATCHDOG_USEC");
 
-        /* lock us into memory */
+        /* Lock us into memory */
         mlockall(MCL_CURRENT|MCL_FUTURE);
 
+        /* Synchronize everything that is not written to disk yet at this point already. This is a good idea so that
+         * slow IO is processed here already and the final process killing spree is not impacted by processes
+         * desperately trying to sync IO to disk within their timeout. */
```

Comment:
The hw watchdog is actually already armed by the original PID 1, right before invoking systemd-shutdown. Hence with this patch applied, if sync() hangs forever the hw watchdog will reboot the system eventually.

So I thought about this a bit more. The shutdown hw watchdog is set to 10min by default. I think that's high enough to simply let sync() do its thing without and additional time limiting code. I mean, the hw watchdog is explicitly for aborting the shutdown when something hangs for too long, and I figure a sync() hanging for 10min plus is actually precisely the condition that it should react to.

Hence I figure the PR the way it is is really fine and should be merged. Makes sense?